iocore/net/SSLNetVConnection.cc

```diff
@@ -1080,7 +1096,7 @@ SSLNetVConnection::sslServerHandShakeEvent(int &err)
     SSLDebugVC(this, "SSL handshake error: %s (%d), errno=%d", SSLErrorName(ssl_error), ssl_error, err);
 
     // start a blind tunnel if tr-pass is set and data does not look like ClientHello
-    char *buf = handShakeBuffer->buf();
+    char *buf = handShakeBuffer ? handShakeBuffer->buf() : NULL;
```

Comment:
Most likely handShakeBuffer is not null yet since we can only make a decision on bad protocol on the client hello (first read).  But this seems like a pretty low overhead protection check.